nearcore/src/lib.rs

```diff
@@ -294,7 +294,7 @@ pub fn start_with_config_and_synchronization(
         move |_ctx| {
             PeerManagerActor::new(
                 time::Clock::real(),
-                store,
+                store.into_inner(),
```

Comment:
are you sure you want to unwrap here, rather than pass a raw DB as an argument to start_with_config_and_synchronization and wrap it for other DB consumers?

Reply:
start_with_config_and_synchronization opens the storage in the first place and in first step it passes it to NightshadeRuntime which does need the whole Store object.